rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend

```diff
@@ -92,45 +104,96 @@ class TclLauncherUi implements Launcher {
 			launcher.class.simpleName, elementId, project)
 		progressRunner.run([ monitor |
 			monitor.beginTask("Test execution: " + elementId, IProgressMonitor.UNKNOWN)
-			val result = launcher.launchTest(selection, project, elementId, monitor, options)
+			val testCasesCommaList = createTestCasesCommaList(selection)
+			val result = launcher.launchTest(testCasesCommaList, project, monitor, options)
 			project.refreshLocal(IProject.DEPTH_INFINITE, monitor)
-			if (result.expectedFile == null) {
+			if (result.expectedFileRoot == null) {
 				logger.error("resulting expectedFile must not be null")
 			} else {
-				safeUpdateJunitTestView(elementId, result.expectedFile)
+				safeUpdateJunitTestView(elementId, result.expectedFileRoot, project.name)
 			}
 			monitor.done
 		])
 		return true
 	}
 
+	def String createTestCasesCommaList(IStructuredSelection selection) {
+		if (launchShortcutUtil == null) {
+			launchShortcutUtil = tclInjectorProvider.get.getInstance(LaunchShortcutUtil)
```

Comment:
Yes I see the problem but we could use constructor injection to get the `TclInjectorProvider` injected in the constructor and then retrieve the `launchShortcutUtil` with the tcl injector.